packages/git/src/browser/git-repository-provider.ts

```diff
@@ -185,11 +186,12 @@ export class GitRepositoryProvider {
         }
     }
 
-    protected registerScmProvider(repository: Repository): void {
+    protected async registerScmProvider(repository: Repository): Promise<void> {
         const provider = this.scmProviderFactory({ repository });
+        const branch = await this.git.branch(repository, { type: 'current' });
         this.scmService.registerScmProvider(provider, {
             input: {
-                placeholder: 'Message (press {0} to commit)',
+                placeholder: `Message (${OS.type() === OS.Type.OSX ? '⌘' : 'press '}{0} to commit ${branch ? 'on \'' + branch.name + '\'' : ''})`,
```

Comment:
one should use [Keybindings.acceleratorFor](https://github.com/eclipse-theia/theia/blob/98f12a9fcc3a43c2b5d224e186f871e66a1e11b7/packages/core/src/browser/keybinding.ts#L307)  to create a human readable representation